compiler/runtime/OMRRuntimeAssumptions.hpp

```diff
@@ -148,6 +148,42 @@ class SentinelRuntimeAssumption : public OMR::RuntimeAssumption
    virtual void     dumpInfo() {};
    }; // TR::SentinelRuntimeAssumption
 
-}
+class PatchNOPedGuardSite : public OMR::LocationRedirectRuntimeAssumption
+   {
+   protected:
+   PatchNOPedGuardSite(TR_PersistentMemory *pm, uintptrj_t key, TR_RuntimeAssumptionKind kind,
+                       uint8_t *location, uint8_t *destination)
+      : OMR::LocationRedirectRuntimeAssumption(pm, key), _location(location), _destination(destination) {}
+
+   public:
+   static  void compensate(bool isSMP, uint8_t *loc, uint8_t *dest);
```

Comment:
I will create another pull request to move over the implementation. I probably need to do a rebase if you are going to merge this, please let me know.